addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -0,0 +1,925 @@
+
+            // "deviceType":"ALEXA_CURRENT_DEVICE_TYPE"
+            String deviceType = "\"deviceType\":\"ALEXA_CURRENT_DEVICE_TYPE\"";
+            String newDeviceType = "\"deviceType\":\"" + device.deviceType + "\"";
+            sequenceJson = sequenceJson.replace(deviceType.subSequence(0, deviceType.length()),
+                    newDeviceType.subSequence(0, newDeviceType.length()));
+
+            // "deviceSerialNumber":"ALEXA_CURRENT_DSN"
+            String deviceSerial = "\"deviceSerialNumber\":\"ALEXA_CURRENT_DSN\"";
+            String newDeviceSerial = "\"deviceSerialNumber\":\"" + device.serialNumber + "\"";
+            sequenceJson = sequenceJson.replace(deviceSerial.subSequence(0, deviceSerial.length()),
+                    newDeviceSerial.subSequence(0, newDeviceSerial.length()));
+
+            // "customerId": "ALEXA_CUSTOMER_ID"
+            String customerId = "\"customerId\":\"ALEXA_CUSTOMER_ID\"";
+            String newCustomerId = "\"customerId\":\"" + device.deviceOwnerCustomerId + "\"";
+            sequenceJson = sequenceJson.replace(customerId.subSequence(0, customerId.length()),
+                    newCustomerId.subSequence(0, newCustomerId.length()));
+
+            // "locale": "ALEXA_CURRENT_LOCALE"
+            String locale = "\"locale\":\"ALEXA_CURRENT_LOCALE\"";
+            String newlocale = StringUtils.isNotEmpty(deviceLocale) ? "\"locale\":\"" + deviceLocale + "\""
+                    : "\"locale\":null";
+            sequenceJson = sequenceJson.replace(locale.subSequence(0, locale.length()),
+                    newlocale.subSequence(0, newlocale.length()));
+
+            request.sequenceJson = sequenceJson;
+
+            String requestJson = gson.toJson(request);
+            makeRequest("POST", alexaServer + "/api/behaviors/preview", requestJson, true, true, null);
+        } else {
+            logger.warn("Routine {} not found", utterance);
+        }
+    }
+
+    public JsonAutomation[] getRoutines() throws IOException, URISyntaxException {
+        String json = makeRequestAndReturnString(alexaServer + "/api/behaviors/automations");
+        JsonAutomation[] result = parseJson(json, JsonAutomation[].class);
+        return result;
+    }
+
+    public JsonFeed[] getEnabledFlashBriefings() throws IOException, URISyntaxException {
+        String json = makeRequestAndReturnString(alexaServer + "/api/content-skills/enabled-feeds");
+        JsonEnabledFeeds result = parseJson(json, JsonEnabledFeeds.class);
+        JsonFeed[] enabledFeeds = result.enabledFeeds;
+        if (enabledFeeds != null) {
+            return enabledFeeds;
+        }
+        return new JsonFeed[0];
+    }
+
+    public void setEnabledFlashBriefings(JsonFeed[] enabledFlashBriefing) throws IOException, URISyntaxException {
+        JsonEnabledFeeds enabled = new JsonEnabledFeeds();
+        enabled.enabledFeeds = enabledFlashBriefing;
+        GsonBuilder gsonBuilder = new GsonBuilder();
```

Comment:
This is potentially expensive - you should instantiate it only once as a field in that class.